src/gpuarray_blas_cuda_cublas.c

```diff
@@ -275,6 +278,39 @@ static void teardown(gpucontext *c) {
   ctx->blas_handle = NULL;
 }
 
+static const char *error(gpucontext *c) {
+  cuda_context *ctx = (cuda_context *)c;
+  blas_handle *handle = (blas_handle *)ctx->blas_handle;
+
+  if (handle != NULL) {
+    switch (handle->err) {
+    case CUBLAS_STATUS_SUCCESS:
+      return "(cublas) Operation completed successfully.";
+    case CUBLAS_STATUS_NOT_INITIALIZED:
+      return "(cublas) Library not initialized.";
+    case CUBLAS_STATUS_ALLOC_FAILED:
+      return "(cublas) GPU ressource allocation failed.";
+    case CUBLAS_STATUS_INVALID_VALUE:
+      return "(cublas) Invalid value.";
+    case CUBLAS_STATUS_ARCH_MISMATCH:
+      return "(cublas) Operation not supported by device.";
+    case CUBLAS_STATUS_MAPPING_ERROR:
+      return "(cublas) Mapping error.";
+    case CUBLAS_STATUS_EXECUTION_FAILED:
+      return "(cublas) Execution failed.";
+    case CUBLAS_STATUS_INTERNAL_ERROR:
+      return "(cublas) Internal error.";
+    case CUBLAS_STATUS_NOT_SUPPORTED:
+      return "(cublas) Unsupported functionality.";
+    case CUBLAS_STATUS_LICENSE_ERROR:
+      return "(cublas) License error.";
+    default:
+      return "(cublas) Unknown error.";
```

Comment:
I don't understand, why can't you build a string and return the string you build? Is this because you fear you can't allocate memory on the CPU for the error string?